src/fty_alert_list_server.cc

```diff
@@ -153,11 +153,17 @@ s_handle_stream_deliver(mlm_client_t *client, zmsg_t** msg_p, zhash_t *expiratio
         cursor = (fty_proto_t *) zlistx_next(alerts);
     }
     if (!found) {
+        // Record creation time
+        fty_proto_aux_insert(newAlert, "ctime", "%" PRIu64, fty_proto_time(newAlert));
+
         zlistx_add_end(alerts, newAlert);
         cursor = (fty_proto_t *) zlistx_last(alerts);
         alertsLastSent[cursor] = 0;
         s_set_alert_lifetime(expirations, newAlert);
     } else {
+        // Append creation time to new alert
+        fty_proto_aux_insert(newAlert, "ctime", "%" PRIu64, fty_proto_aux_number(cursor, "ctime", 0));
```

Comment:
I think you could have used fty_proto_time(cursor) which might have been a bit faster, but this should be fine